command/rubyscript/rubyscript_test.go

```diff
@@ -69,30 +72,48 @@ func Test_ensureTmpDir(t *testing.T) {
 func TestBundleInstallCommand(t *testing.T) {
 	t.Log("bundle install gems")
 	{
+		mockFactory := new(mocks.Factory)
+		mockCommand := new(mocks.Command)
+		mockCommand.On("Run").Return(nil)
+		mockFactory.On("Create", mock.Anything, mock.Anything, mock.Anything).Return(mockCommand)
+		temporaryFactory = mockFactory
+
 		runner := New(rubyScriptWithGemContent)
 		require.NotNil(t, runner)
 
 		bundleInstallCmd, err := runner.BundleInstallCommand(gemfileContent, gemfileLockContent)
 		require.NoError(t, err)
 
-		cmd := bundleInstallCmd.GetCmd()
-
-		require.Equal(t, filepath.Base(cmd.Path), "bundle")
-		require.Equal(t, 3, len(cmd.Args))
-		require.Equal(t, "bundle", cmd.Args[0])
-		require.Equal(t, "install", cmd.Args[1])
-
-		gemfileFlag := cmd.Args[2]
-		split := strings.Split(gemfileFlag, "=")
-		require.Equal(t, 2, len(split))
-		require.Equal(t, "--gemfile", split[0])
-		require.Equal(t, "Gemfile", filepath.Base(split[1]))
+		called := false
+		for _, call := range mockFactory.Calls {
+			if call.Method == "Create" && len(call.Arguments) == 3 && call.Arguments[0] == "bundle" {
+				s, ok := call.Arguments[1].([]string)
+				if ok && len(s) == 2 {
+					if s[0] == "install" {
+						split := strings.Split(s[1], "=")
+						if len(split) == 2 && split[0] == "--gemfile" && filepath.Base(split[1]) == "Gemfile" {
+							called = true
+						}
+					}
+				}
+			}
+		}
+		if !called {
+			assert.Fail(t, "No invocation of Create with right arguments")
+		}
 
 		require.NoError(t, bundleInstallCmd.Run())
```

Comment:
With these changes we no longer run the actual ruby commands in the unit tests. If you think that is still needed we can change the implementation

Reply:
It is... acceptable.